api/src/utils/get-schema.ts

```diff
@@ -31,33 +33,44 @@ export async function getSchema(options?: {
 	const database = options?.database || getDatabase();
 	const schemaInspector = createInspector(database);
```

Comment:
As a small additional optimization, we can move these into the respective places, so they are only called when required

Reply:
These are used in the if statement immediately following:

https://github.com/directus/directus/blob/c70874fc757e1584db0974c87f90df06e5886898/api/src/utils/get-schema.ts#L36-L38

Reply:
Yeah, and only used inside it as well as further down in the `try` statement. So no need to use unnecessary resources in my opinion:

```ts
	if (options?.bypassCache || env['CACHE_SCHEMA'] === false) {
		const database = options?.database || getDatabase();
		const schemaInspector = createInspector(database);
		return await getDatabaseSchema(database, schemaInspector);
	}

// ...

	try {
		const database = options?.database || getDatabase();
		const schemaInspector = createInspector(database);
		const schema = await getDatabaseSchema(database, schemaInspector);
```